proto/buf/alpha/registry/v1alpha1/studio_request.proto

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+// StudioRequestService exposes the functionality to manage favorited Requests
+// from Studio.
+service StudioRequestService {
+  // CreateStudioRequest registers a favorite Studio Requests to the caller's
+  // BSR profile.
+  rpc CreateStudioRequest(CreateStudioRequestRequest) returns (CreateStudioRequestResponse) {}
+  // RenameStudioRequest renames an existing Studio Request.
+  rpc RenameStudioRequest(RenameStudioRequestRequest) returns (RenameStudioRequestResponse) {}
+  // DeleteStudioRequest removes a favorite Studio Request from the caller's BSR
+  // profile.
+  rpc DeleteStudioRequest(DeleteStudioRequestRequest) returns (DeleteStudioRequestResponse) {}
+  // ListStudioRequests shows the caller's favorited Studio Requests.
+  rpc ListStudioRequests(ListStudioRequestsRequest) returns (ListStudioRequestsResponse) {}
+}
+
+// CreateStudioRequestRequest is the proto request representation of the
+// CreateStudioRequest method. IDs and timestamps are automatically generated,
+// BSR Repository information is taken from the `repository_id`, and the
+// author is taken from the request authentication.
+message CreateStudioRequestRequest {
+  // BSR repository ID this request is linked to.
+  string repository_id = 1;
+  // The friendly name of this request.
+  string name = 2;
+  // The target base URL of this request.
+  string target_base_url = 3;
+  // The target service name as defined in the BSR repository.
+  string service = 4;
+  // The target RPC method name as defined in the BSR repository.
+  string method = 5;
+  // The request body as defined in Studio.
+  string request_body = 6;
+  // The request headers as defined in Studio.
+  map<string, string> request_headers = 7;
+  // The response body as seen in Studio.
+  string response_body = 8;
+  // The response headers as seen in Studio.
+  map<string, string> response_headers = 9;
```

Comment:
Looking at https://studio.buf.build/, there are two other fields that I think we may need:
- Under options we mark if the request is to be made with cookies or not
<img width="1075" alt="image" src="https://user-images.githubusercontent.com/11449728/183086304-764b769d-53b8-48d3-933e-637372d389eb.png">
- What protocol was it made with
<img width="1035" alt="image" src="https://user-images.githubusercontent.com/11449728/183086384-17396a67-1a08-4f28-937b-0cf9b28d6b80.png">
- What agent was associated with the request
<img width="900" alt="image" src="https://user-images.githubusercontent.com/11449728/183086460-149f478b-f42c-40c1-8465-0dc4e5509479.png">

Perhaps this was mentioned somewhere in notion about how we are handling these fields, but I would assume we should handle them?

Reply:
Nice point! No those weren't mentioned. I'm thinking, to account for these fields and to support potential new ones, just add `map<string, string> meta`? And even move the `response_body` and `response_headers` there (I had no strong opinions on storing responses values)? even timestamps and duration of requests if we're feeling that'd bring some value later?

Reply:
I am up for a generic meta if necessary. I've used it elsewhere before, though it does lose any type safety for those fields. Kinda depends on how dogmatic we want to be about making sure all our APIs are type safe. I could argue that if we're trying to be the "replace JSON" company, then we should stick to it and make all these fields very explicit, even if they all get squished and stored as generic key value pairs on the receiving end.

Reply:
You're right, I'll add them separately.